pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -276,89 +257,96 @@ contract IndexPool is BaseWeightedPool, ReentrancyGuard {
         _require(normalizedSum == FixedPoint.ONE, Errors.NORMALIZED_WEIGHT_INVARIANT);
     }
 
-    function _interpolateWeight(
-        uint256 startWeight,
-        uint256 endWeight,
-        uint256 pctProgress
-    ) private pure returns (uint256) {
+    function _interpolateWeight(bytes32 tokenData, uint256 pctProgress) private pure returns (uint256 finalWeight) {
+        uint256 startWeight = tokenData.decodeUint64(_START_WEIGHT_OFFSET).uncompress64();
+        uint256 endWeight = tokenData.decodeUint32(_END_WEIGHT_OFFSET).uncompress32();
+
         if (pctProgress == 0 || startWeight == endWeight) return startWeight;
         if (pctProgress >= FixedPoint.ONE) return endWeight;
 
         if (startWeight > endWeight) {
             uint256 weightDelta = pctProgress.mulDown(startWeight - endWeight);
-            return startWeight.sub(weightDelta);
+            return startWeight - weightDelta;
```

Comment:
it's better to use sub and add here, it's Solidity 0.7.0, if I remember correctly Safe arithmetics is a feature of 0.8.0^